frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -1321,3 +1325,15 @@ func useFileOp(args map[string]string, caps *apicaps.CapSet) bool {
 	}
 	return enabled && caps != nil && caps.Supports(pb.CapFileBase) == nil
 }
+
+func appendArgs(cmd string, buildArgs []instructions.KeyValuePairOptional, definedArgs []string) string {
+	var prefixArgs string
+	for _, arg := range definedArgs {
+		for _, v := range buildArgs {
+			if v.Key == arg {
+				prefixArgs = prefixArgs + " " + arg + "=" + *v.Value
+			}
+		}
+	}
+	return cmd + prefixArgs
```

Comment:
These should go to the beginning, not end. I think you can `s/definedArgs/prefixArgs/` and append directly to `cmd`